### PR TITLE
fixed injvm export and refer

### PR DESCRIPTION
**Pull request by Leishunyu**
## What is the purpose of the change

https://github.com/apache/incubator-dubbo/issues/3638
Fixed bug in this issue

## Brief changelog
Modified the original error logic

## Verifying this change
        ServiceConfig<DemoServiceImpl> service = new ServiceConfig<>();
        ApplicationConfig applicationConfig = new ApplicationConfig("dubbo-demo-api-provider");
        service.setVersion("1.0.1");
        service.setProtocol(new ProtocolConfig("injvm"));
        service.setApplication(applicationConfig);
        //service.setScope("local");
        service.setInterface(DemoService.class);
        service.setRef(new DemoServiceImpl());
        service.export();
        ReferenceConfig<DemoService> reference = new ReferenceConfig<>();
        reference.setVersion("1.0.1");
        reference.setProtocol("injvm");
        //reference.setScope("local");
        reference.setApplication(applicationConfig);
        reference.setInterface(DemoService.class);
        DemoService serviceImpl = reference.get();
       serviceImpl.sayHello("111");

Follow this checklist to help us incorporate your contribution quickly and easily:

- [x] Make sure there is a [GITHUB_issue](https://github.com/apache/incubator-dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GITHUB issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [ ] Format the pull request title like `[Dubbo-XXX] Fix UnknownException when host config not exist #XXX`. Each commit in the pull request should have a meaningful subject line and body.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/apache/incubator-dubbo/tree/master/dubbo-test).
- [x] Run `mvn clean install -DskipTests=false` & `mvn clean test-compile failsafe:integration-test` to make sure unit-test and integration-test pass.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/incubator-dubbo/wiki/Software-donation-guide).

### CODE REVIEW

**Comment by Leishunyu** [comment]
@beiwei30  Can you help me see?

**Comment by CrazyHZM** [comment]
Can you add some unit cases?

**Comment by Leishunyu** [comment]
> Can you add some unit cases?
The previous unit test has been covered

**Comment by Leishunyu** [comment]
@beiwei30 can you help me review

**Comment by beiwei30** [comment]
on it.

File: dubbo-config/dubbo-config-api/src/main/java/org/apache/dubbo/config/ServiceConfig.java [hunk 1]
```diff
@@ -324,6 +327,15 @@ public void checkAndUpdateSubConfigs() {
         checkMock(interfaceClass);
     }
 
+    /**
+     * Determine if it is injvm
+     *
+     * @return
+     */
+    private boolean isJvm() {
```
Comment (beiwei30) [inline dubbo-config/dubbo-config-api/src/main/java/org/apache/dubbo/config/ServiceConfig.java]: pls. change method name from **isJvm** to **isOnlyInJvm**

Comment (Leishunyu) [reply dubbo-config/dubbo-config-api/src/main/java/org/apache/dubbo/config/ServiceConfig.java]: get

**Pull request merged**